app/src/main/java/org/mozilla/fenix/search/SearchDialogFragment.kt

```diff
@@ -343,19 +352,7 @@ class SearchDialogFragment : AppCompatDialogFragment(), UserInteractionHandler {
             view.hideKeyboard()
             toolbarView.view.clearFocus()
 
-            if (requireContext().settings().shouldShowCameraPermissionPrompt) {
-                qrFeature.get()?.scan(binding.searchWrapper.id)
-            } else {
-                if (requireContext().isPermissionGranted(Manifest.permission.CAMERA)) {
-                    qrFeature.get()?.scan(binding.searchWrapper.id)
-                } else {
-                    interactor.onCameraPermissionsNeeded()
-                    resetFocus()
-                    view.hideKeyboard()
```

Comment:
Drive-by review: Are we certain, we don't need to dismiss the keyboard as well?

Reply:
Looking at it there is a specific call to close the keyboard when dismissing the dialog : 
https://github.com/mozilla-mobile/fenix/blob/5cce4b5f15ffcbfa70079318f284c67de6d8bb6b/app/src/main/java/org/mozilla/fenix/search/SearchDialogFragment.kt#L548

Comments of `hideDeviceKeyboard` method reference what appear to be a different use case for the dialog, will look at it.